### PR TITLE
Prepare EVM Support Changes

### PR DESCRIPTION
**Pull request by Bushstar**
Adds all CPP side changes from feature/evm to master so that there will be less diffs when merging feature/evm into master.

### CODE REVIEW

**Review by Mixa84 — APPROVED** [review]

File: src/masternodes/mn_checks.cpp [hunk 1]
```diff
@@ -3855,13 +3855,11 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
         if (obj.evmTx.size() > static_cast<size_t>(EVM_TX_SIZE))
             return Res::Err("evm tx size too large");
 
-        if (!evm_validate_raw_tx(HexStr(obj.evmTx))) {
+        if (!evm_prevalidate_raw_tx(HexStr(obj.evmTx)))
             return Res::Err("evm tx failed to validate");
-        }
 
-        if (!evm_queue_tx(evmContext, HexStr(obj.evmTx))) {
+        if (!evm_queue_tx(evmContext, HexStr(obj.evmTx), tx.GetHash().ToArrayReversed()))
```
Comment (sieniven) [inline src/masternodes/mn_checks.cpp:3861]: Seems like the changes in commit ba4eac0622 (https://github.com/DeFiCh/ain/pull/2025/commits/ba4eac0622ccffadf100b01d39c62472a09c3212) in PR #2025 is not reflected here.

Comment (Bushstar) [reply src/masternodes/mn_checks.cpp:3861]: Good spot. I'm updating the CPP changes and will look for this.

File: src/masternodes/mn_checks.cpp [hunk 2]
```diff
@@ -3920,7 +3920,7 @@ Res ValidateTransferDomain(const CTransaction &tx,
 
         CTxDestination dest;
 
-        // Soruce validation
```
Comment (Bushstar) [inline src/masternodes/mn_checks.cpp:3925]: Ignore stuff like this. This is now a difference between feature/evm and master. We can sort this out after merging feature/evm.

File: src/masternodes/mn_checks.cpp (hunk 1 above)
Comment (Bushstar) [reply src/masternodes/mn_checks.cpp]: This is now included. It was merged after this PR was created so was missing initially.

**Review by sieniven — APPROVED** [review]

**Pull request merged**